admob/src/include/firebase/admob/types.h

```diff
@@ -309,6 +447,66 @@ struct BoundingBox {
   int y;
 };
 
+/// Information about an ad response.
+class ResponseInfo {
+ public:
+  /// Constructor creates an uninitialized ResponseInfo.
+  ResponseInfo();
+
+  /// Gets the AdaptorReponseInfo objects for the ad response.
+  ///
+  /// @return a vector of AdapterResponseInfo objects containing metadata for
+  ///   each adapter included in the ad response.
+  const std::vector<AdapterResponseInfo>& adapter_responses() const {
+    return adapter_responses_;
+  }
+
+  /// A class name that identifies the ad network that returned the ad.
+  /// Returns an empty string if the ad failed to load.
+  const std::string& mediation_adapter_class_name() const {
+    return mediation_adapter_class_name_;
+  }
+
+  /// Gets the response ID string for the loaded ad.  Returns an empty
+  /// string if the ad fails to load.
+  const std::string& response_id() const { return response_id_; }
+
+  /// Gets a log friendly string version of this object.
+  const std::string& ToString() const { return to_string_; }
+
+ private:
+  friend class LoadAdResult;
+
+  explicit ResponseInfo(const ResponseInfoInternal& internal);
+
+  std::vector<AdapterResponseInfo> adapter_responses_;
+  std::string mediation_adapter_class_name_;
+  std::string response_id_;
+  std::string to_string_;
+};
+
+/// @brief Information about why an ad load operation failed.
+class LoadAdResult : public AdResult {
```

Comment:
One more question - the only thing this adds on top of AdResult is the ResponseInfo, right? Is there any reason that shouldn't just be in the base AdResult and avoid subclassing at all? Seems like a lot of mental overhead to subclass AdResult just to add one field.